types/index.d.ts

```diff
@@ -0,0 +1,344 @@
+	/**
+	 * _For creating indexed PNGs._ The index of the background color. Defaults
+	 * to 0.
+	 */
+	backgroundIndex?: number
+	/** pixels per inch */
+	resolution?: number
+}
+
+export class PngStream extends Readable {}
+
+export interface JpegConfig {
+	/** Specifies the quality, between 0 and 1. Defaults to 0.75. */
+	quality?: number
+	/** Enables progressive encoding. Defaults to `false`. */
+	progressive?: boolean
+	/** Enables 2x2 chroma subsampling. Defaults to `true`. */
+	chromaSubsampling?: boolean
+}
+
+export class JpegStream extends Readable {}
+
+export interface PdfConfig {
+	title?: string
+	author?: string
+	subject?: string
+	keywords?: string
+	creator?: string
+	creationDate?: Date
+	modDate?: Date
+}
+
+export class PdfStream extends Readable {}
+
+export interface NodeCanvasRenderingContext2DSettings {
+	alpha?: boolean
+	pixelFormat?: 'RGBA32' | 'RGB24' | 'A8' | 'RGB16_565' | 'A1' | 'RGB30'
+}
+
+export class Canvas {
+	width: number
+	height: number
+
+	/** _Non standard._ The type of the canvas. */
+	readonly type: 'image'|'pdf'|'svg'
+
+	// This is a getter, but it's non-standard and I don't know why we export it.
+	// readonly stride: number;
+
+	/** Constant used in PNG encoding methods. */
+	readonly PNG_NO_FILTERS: number
+	/** Constant used in PNG encoding methods. */
+	readonly PNG_ALL_FILTERS: number
+	/** Constant used in PNG encoding methods. */
+	readonly PNG_FILTER_NONE: number
+	/** Constant used in PNG encoding methods. */
+	readonly PNG_FILTER_SUB: number
+	/** Constant used in PNG encoding methods. */
+	readonly PNG_FILTER_UP: number
+	/** Constant used in PNG encoding methods. */
+	readonly PNG_FILTER_AVG: number
+	/** Constant used in PNG encoding methods. */
+	readonly PNG_FILTER_PAETH: number
+
+	constructor(width: number, height: number, type?: 'pdf'|'svg')
+
+	getContext(contextId: '2d', contextAttributes?: NodeCanvasRenderingContext2DSettings): NodeCanvasRenderingContext2D
+
+	/**
+	 * For image canvases, encodes the canvas as a PNG. For PDF canvases,
+	 * encodes the canvas as a PDF. For SVG canvases, encodes the canvas as an
+	 * SVG.
+	 */
+	toBuffer(cb: (err: Error|null, result: Buffer) => void): void
+	toBuffer(cb: (err: Error|null, result: Buffer) => void, mimeType: 'image/png', config?: PngConfig): void
+	toBuffer(cb: (err: Error|null, result: Buffer) => void, mimeType: 'image/jpeg', config?: JpegConfig): void
+
+	/**
+	 * For image canvases, encodes the canvas as a PNG. For PDF canvases,
+	 * encodes the canvas as a PDF. For SVG canvases, encodes the canvas as an
+	 * SVG.
+	 */
+	toBuffer(): Buffer
+	toBuffer(mimeType: 'image/png', config?: PngConfig): Buffer
+	toBuffer(mimeType: 'image/jpeg', config?: JpegConfig): Buffer
+	toBuffer(mimeType: 'application/pdf', config?: PdfConfig): Buffer
+
+	/**
+	 * Returns the raw ARGB data without encoding in native-endian byte order,
+	 * top-to-bottom.
+	 */
+	toBuffer(mimeType: 'raw'): Buffer
+
+	createPNGStream(config?: PngConfig): PngStream
```

Comment:
https://github.com/Automattic/node-canvas/pull/1330/files#diff-c472f8ca9fb190272d5098f1ccd7bd41R32

requires definition pngStream , jpegStream , pdfStream method typing

Reply:
Ahh I thought you were asking for those classes to be defined, now I see you meant the method aliases. Those method aliases are present for backwards compatibility; we standardized on Node.js core's idiom of `create____` in #1152. I'd rather not type them since they're not otherwise documented, but can add them if you think it's important.

Reply:
I think it should be backwards compatible.

Maybe you can learn how to React.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e1f9647734bd990f49b0aca0f44acd4d55fe1e93/types/react/index.d.ts#L456-L462